### ISSUE TITLE
bug in gradient accumulation example

### ISSUE THREAD

**Issue by nilesh2797** [huggingface/accelerate#435]
https://github.com/huggingface/accelerate/blob/450d51ce0191020408bd3481bde85fe1dabaf289/examples/by_feature/gradient_accumulation.py#L160

Shouldn't this condition be `step % gradient_accumulation_steps != 0` because we want to avoid gradient averaging at each step other than `gradient_accumulation_steps` step?

**Comment by muellerzr** [comment]
Traditionally it's step+1 since it starts at 0 if you look at examples 😄 

E.g. see here: https://gist.github.com/mcarilli/bf013d2d2f4b4dd21ade30c9b52d5e2e

(So it just looks backwards)

**Comment by nilesh2797** [comment]
Let's say `gradient_accumulation_steps = 5`, then In the current code `optimizer.step()` is getting called in step `1,2,3,4,6,7,8,9,11,...` instead it should be getting called in step `0,5,10,15,20,...` right? Basically it's in the opposite logical block

**Comment by muellerzr** [comment]
Should be fixed now thanks to #459 @nilesh2797

**Comment by nilesh2797** [comment]
Thanks for the fix!
